pkg/generate/app/dockerimagelookup.go

```diff
@@ -41,13 +40,18 @@ type DockerClientSearcher struct {
 }
 
 // Search searches all images in local docker server for images that match terms
-func (r DockerClientSearcher) Search(terms ...string) (ComponentMatches, error) {
+func (r DockerClientSearcher) Search(precise bool, terms ...string) (ComponentMatches, []error) {
 	componentMatches := ComponentMatches{}
 	errs := []error{}
 	for _, term := range terms {
+		if term == "__dockerimage_fail" {
+			errs = append(errs, fmt.Errorf("unable to find the specified docker image: %s", term))
+			continue
+		}
+
 		ref, err := imageapi.ParseDockerImageReference(term)
 		if err != nil {
-			return nil, err
+			continue
```

Comment:
passing a term that is not a valid docker image reference is not an error, it is simply "there is nothing that could possibly match this".  If we didn't parse the reference, or had partial matching, we could.